plugins/couchbase/CMakeLists.txt

```diff
@@ -25,37 +25,68 @@
 project(couchbaseembed)
 
 IF ( COUCHBASEEMBED )
-    ADD_PLUGIN(COUCHBASEEMBED PACKAGES COUCHBASE)
-    IF ( MAKE_COUCHBASEEMBED )
-        set(
-            SRCS
-            couchbaseembed.cpp
+
+    if(APPLE)
+        set(LIBCOUCHBASE_LIB ${CMAKE_CURRENT_BINARY_DIR}/libcouchbase-build/lib/libcouchbase.so)
+        set(LIBCOUCHBASE_LIB_SONAME ${CMAKE_CURRENT_BINARY_DIR}/libcouchbase-build/lib/libcouchbase.so.2)
+        set(LIBCOUCHBASE_LIB_REAL ${CMAKE_CURRENT_BINARY_DIR}/libcouchbase-build/lib/libcouchbase.so.2.0.38)
+    else()
+        set(LIBCOUCHBASE_LIB ${CMAKE_CURRENT_BINARY_DIR}/libcouchbase-build/lib/libcouchbase.so)
+        set(LIBCOUCHBASE_LIB_SONAME ${CMAKE_CURRENT_BINARY_DIR}/libcouchbase-build/lib/libcouchbase.so.2)
+        set(LIBCOUCHBASE_LIB_REAL ${CMAKE_CURRENT_BINARY_DIR}/libcouchbase-build/lib/libcouchbase.so.2.0.38)
+    endif()
+    add_custom_command(
+        OUTPUT ${LIBCOUCHBASE_LIB}
+        COMMAND mkdir -p ${CMAKE_CURRENT_BINARY_DIR}/libcouchbase-build
+        COMMAND cd ${CMAKE_CURRENT_BINARY_DIR}/libcouchbase-build && cmake ${CMAKE_CURRENT_SOURCE_DIR}/libcouchbase 1>/dev/null && make 1>/dev/null
+        COMMENT Building libcouchbase for import)
+    add_custom_target(libcouchbase-build ALL DEPENDS ${LIBCOUCHBASE_LIB})
+    add_library(libcouchbase SHARED IMPORTED)
+    set_property(TARGET libcouchbase PROPERTY IMPORTED_LOCATION ${LIBCOUCHBASE_LIB})
+    add_dependencies(libcouchbase libcouchbase-build)
+
+    set(
+        SRCS
+        couchbaseembed.cpp
         )
 
-        INCLUDE_DIRECTORIES (
-            ${PROJECT_SOURCE_DIR}/libcouchbase-cxx/include
-            ./../../esp/platform #for token serialize/deserialize
-            ./../../system/include
-            ./../../rtl/eclrtl
-            ./../../rtl/include
-            ./../../rtl/nbcd
-            ./../../common/deftype
-            ./../../system/jlib
-            ./../../roxie/roxiemem
+    INCLUDE_DIRECTORIES (
+        ${PROJECT_SOURCE_DIR}/libcouchbase/include
+        ${PROJECT_BINARY_DIR}/libcouchbase-build/generated
+        ${PROJECT_SOURCE_DIR}/libcouchbase-cxx/include
+        ./../../esp/platform #for token serialize/deserialize
```

Comment:
let's not mix path styles. can we use ${PROJECT_SOURCE_DIR} instead of the relative paths?